server/src/main/java/org/opensearch/search/backpressure/stats/SearchBackpressureStats.java

```diff
@@ -46,18 +60,23 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public void writeTo(StreamOutput out) throws IOException {
         searchShardTaskStats.writeTo(out);
         out.writeString(mode.getName());
+        if (Version.CURRENT.onOrAfter(Version.V_3_0_0) && out.getVersion().onOrAfter(Version.V_3_0_0)) {
+            searchTaskStats.writeTo(out);
+        }
     }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         SearchBackpressureStats that = (SearchBackpressureStats) o;
-        return searchShardTaskStats.equals(that.searchShardTaskStats) && mode == that.mode;
+        return (Version.CURRENT.onOrAfter(Version.V_3_0_0)
```

Comment:
In the constructor of `SearchBackpressureStats` [`public SearchBackpressureStats(StreamInput in)`], we are initialising `searchTaskStats` to `null` if the current version is below 3.0 and now if we do not check for the current version in `writeTo()` or `equals()`, it would cause NPE.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/5605#discussion_r1071814766